packages/lesswrong/server/repos/CommentsRepo.ts

```diff
@@ -582,22 +587,137 @@ class CommentsRepo extends AbstractRepo<"Comments"> {
       LEFT JOIN "CommentEvents" ce ON c._id = ce."documentId"
       ORDER BY COALESCE(c."topLevelCommentId", c._id), c."postedAt"
       LIMIT $(maxTotalComments) -- Apply final limit
-    `, { userId, initialCandidateLimit, maxTotalComments });
+    `, { 
+      userId, 
+      initialCandidateLimit, 
+      maxTotalComments,
+      initialCandidateLookbackDaysParam: initialCandidateLookbackDays,
+      commentServedEventRecencyHoursParam: commentServedEventRecencyHours
+    });
 
     return feedCommentsData.map((comment): FeedCommentFromDb => ({
       commentId: comment.commentId,
+      authorId: comment.authorId,
       topLevelCommentId: comment.topLevelCommentId,
       parentCommentId: comment.parentCommentId ?? null,
       postId: comment.postId,
       baseScore: comment.baseScore,
       shortform: comment.shortform ?? null,
       postedAt: comment.postedAt,
-      sources: ['recentComments'], // Assign fixed source
+      sources: ['recentComments'],
       lastServed: null, 
       lastViewed: comment.lastViewed ?? null,
       lastInteracted: comment.lastInteracted ?? null,
     }));
   }
+
+  /**
+   * Fetches consolidated engagement statistics for recently active comment threads.
+   */
+  async getThreadEngagementStatsForRecentlyActiveThreads(
+    userId: string,
+    threadEngagementLookbackDays: number
+  ): Promise<ThreadEngagementStats[]> {
+    const db = this.getRawDb();
+    const threadCandidateLimit = 200; // Hardcoded
+    const lookbackInterval = `${threadEngagementLookbackDays} days`;
+
+    const engagementStats = await db.manyOrNone<ThreadEngagementStats>(`
+      -- CommentsRepo.getThreadEngagementStatsForRecentlyActiveThreads
+      WITH "RecentActiveThreads" AS (
```

Comment:
My guess is that this and the next 3 CTEs can be collapsed into a join (not sure about the 4th, because of the UNION in it)

Reply:
Maybe...I feel like that hurts readability a lot